lib/checker.js

```diff
@@ -60,15 +93,22 @@ Checker.prototype.checkDirectory = function(path) {
             }
 
             return vowFs.stat(fullname).then(function(stat) {
+                var promise;
+
                 if (stat.isDirectory()) {
                     return this.checkDirectory(fullname);
                 }
 
                 if (!this._hasCorrectExtension(fullname)) {
-                    return [];
+                    if (this._configuration.getExtract() === 'never') {
+                        return [];
+                    }
+                    promise = Vow.when(this.extractFile(fullname));
```

Comment:
If `JSHint` did it, it doesn't mean it is a good solution, with `extract = never` by default, user would need to know about this option, with `extract = auto` by default, it would hurt performance.

Whereas if we would extend `fileExtensions` we would support most use-cases performance downside.

> I believe we should do initial solution as simple as possible.

If we would add `extract` option we need to support it for long time.